Objects/longobject.c

```diff
@@ -4565,11 +4586,18 @@ _PyLong_Rshift(PyObject *a, size_t shiftby)
 static PyObject *
 long_lshift1(PyLongObject *a, Py_ssize_t wordshift, digit remshift)
 {
-    /* This version due to Tim Peters */
     PyLongObject *z = NULL;
     Py_ssize_t oldsize, newsize, i, j;
     twodigits accum;
 
+    if (wordshift == 0 && IS_MEDIUM_VALUE(a) &&
+        (a->ob_digit[0] & ~(PyLong_MASK >> remshift)) == 0) {
+        // bypass undefined shift operator behavior
+        stwodigits x = (stwodigits)((twodigits)medium_value(a) << remshift);
+        return _PyLong_FromSTwoDigits(x);
+    }
+
+    /* This version due to Tim Peters */
```

Comment:
This comment is becoming increasingly inaccurate and confusing (exactly *which* part of the code does it apply to?), and curiously specific (since so much else in this module is also due to Tim Peters). @tim-one: would you object to it being dropped?

Reply:
Get rid of it - looks like @serhiy-storchaka added it in 2019, presumably to make it clearer at the time which parts they were adding.

Reply:
I did not add it, I just moved some code before and after this line, and git was confused.

It was added by GvR in 1997 (f2e499b1d760076f519b618105c112495d6f7911).

Reply:
OK! After a quarter century, I expect it's served its purpose & can be retired now 😄